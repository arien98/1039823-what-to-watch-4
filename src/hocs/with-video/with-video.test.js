import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withVideo from "./with-video";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

it(`with button renders correctly`, () => {
  const filmData = {
    title: `Aviator`,
    poster: `../img/aviator.jpg`,
    genre: `Drama`,
    releaseDate: 2010,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  };

  const tree = renderer.create(
      <MockComponentWrapped
        filmData={filmData}
        onFilmCardClick={() => {}}
      />
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
