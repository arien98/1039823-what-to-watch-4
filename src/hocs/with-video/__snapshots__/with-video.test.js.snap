// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with button renders correctly 1`] = `
<div>
  <video
    autoPlay={true}
    className="small-movie-card__video"
    height="175"
    muted={true}
    poster="../img/aviator.jpg"
    width="280"
  >
    <source
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    />
  </video>
</div>
`;
